src/osdc/Objecter.cc

```diff
@@ -1128,6 +1132,10 @@ void Objecter::_scan_requests(
 
   sl.unlock();
 
+  for (auto const& p: cancel_ops) {
+    p.first->cancel_ops(p.second);
+  }
+
   for (list<LingerOp*>::iterator iter = unregister_lingers.begin();
        iter != unregister_lingers.end();
        ++iter) {
```

Comment:
I definitely want this commit. I can use it to interfaces for RADOS_unleashed that just take byte ranges or spans or what-have-you without the caller needing to provide a bufferlist at all.